flamedisx/inference.py

```diff
@@ -90,7 +94,8 @@ def __init__(
 
         # Create sources. Have to copy data, it's modified by set_data
         self.sources = {
-            sname: sclass(data[self.d_for_s[sname]].copy(),
+            sname: sclass(data=None if data[self.d_for_s[sname]] is None \
```

Comment:
Maybe you could look into the formatting here

Reply:
If only we had Python 3.8 :) 
```python
data=d if (d := data[self.d_for_s[sname]]) is None else d.copy()
```
but I'll remove the backslash for you.